src/Nethermind/Nethermind.Synchronization/SnapSync/SnapProviderHelper.cs

```diff
@@ -17,6 +17,8 @@ namespace Nethermind.Synchronization.SnapSync
 {
     public static class SnapProviderHelper
     {
+        private const int ExtensionRlpChildIndex = 1;
```

Comment:
Shouldnt it always be 0?

Reply:
Ow, no, sometimes its 1 its 0. That can't be good.

Reply:
Some methods add 1 if extension (converting 0 -> 1); is second element in rlp for extension (first being key); these methods doesn't add one